Emphasize use of state variable for custom devices

tktf50 opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

This documentation update is primarily to emphasize the need to use a state variable (not an integer variable) when configuring a custom component. Also fixed a few grammatical errors.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: #20558

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

Commit:
Emphasize use of state variable for custom devices

This documentation update is primarily to emphasize the need to use a state variable (not an integer variable) when configuration a custom component. Also fixed a few grammatical errors.

homeassistant commented:
Hi @TKTF50, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#20559).

Once you do that we will be able to review and accept this pull request.

Thanks!

frenck commented on source/_integrations/isy994.markdown:
```diff
@@ -330,13 +330,19 @@ A device is created by creating a directory, with the name for the device, under
+The *status* program requires that you create a state variable with the name of your choice. Note that you must use a state variable, not an integer variable. This variable will store the actual status of the new device and will be updated by the *action* program.
+
+<p class='img'>
+  <img src='/images/isy994/isy994_SwitchVariableExample.png' />
```
- Please lower care the file name and use snake case filename styling.
- Please use Markdown to embed the image.

bdraco reviewed (dismissed):
Please see comment above. Thanks 👍 

https://github.com/home-assistant/home-assistant.io/pull/20559#discussion_r762152563

Commit:
Emphasize use of state variable for custom devices

This documentation update is primarily to emphasize the need to use a state variable (not an integer variable) when configuring a custom component. Also fixed a few grammatical errors.

frenck reviewed (approved):
Thanks, @TKTF50 👍
